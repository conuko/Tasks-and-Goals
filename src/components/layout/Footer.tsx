import GitHub from "../../assets/images/icons8-github-48.png";
import LinkedIn from "../../assets/images/icons8-linkedin-48.png";

const Footer = () => {
  return (
    <footer className="footer">
      <div className="footer__child">
        <div>
          <p>
            <strong>Shortlist</strong> by{" "}
            <a
              className="footer__link"
              href="https://github.com/conuko"
              target="_blank"
            >
              @conuko
            </a>
            <br />
            <a href="http://www.github.com/conuko" target="_blank">
              <img className="footer__icon" src={GitHub} alt="GitHub" />
            </a>
            <a
              href="https://www.linkedin.com/in/constantin-unterkofler"
              target="_blank"
            >
              <img className="footer__icon" src={LinkedIn} alt="LinkedIn" />
            </a>
          </p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
