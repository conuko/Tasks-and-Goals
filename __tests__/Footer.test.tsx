import "@testing-library/jest-dom";
import { render, screen } from "../src/utils/test-utils";
import Footer from "../src/components/layout/Footer";

test("Should render footer", () => {
  const { baseElement } = render(<Footer />);
  const footerHeader = screen.getByText("Shortlist");
  const githubLink = screen.getByAltText("GitHub");
  const linkedinLink = screen.getByAltText("LinkedIn");

  expect(baseElement).toMatchSnapshot();
  expect(footerHeader).toBeInTheDocument();
  expect(githubLink).toBeInTheDocument();
  expect(linkedinLink).toBeInTheDocument();
});
