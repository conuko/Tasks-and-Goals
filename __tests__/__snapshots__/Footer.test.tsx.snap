// Vitest Snapshot v1

exports[`Should render footer 1`] = `
<body>
  <div>
    <footer
      class="footer"
    >
      <div
        class="footer__child"
      >
        <div>
          <p>
            <strong>
              Shortlist
            </strong>
             by
             
            <a
              class="footer__link"
              href="https://github.com/conuko"
              target="_blank"
            >
              @conuko
            </a>
            <br />
            <a
              href="http://www.github.com/conuko"
              target="_blank"
            >
              <img
                alt="GitHub"
                class="footer__icon"
                src="/src/assets/images/icons8-github-48.png"
              />
            </a>
            <a
              href="https://www.linkedin.com/in/constantin-unterkofler"
              target="_blank"
            >
              <img
                alt="LinkedIn"
                class="footer__icon"
                src="/src/assets/images/icons8-linkedin-48.png"
              />
            </a>
          </p>
        </div>
      </div>
    </footer>
  </div>
</body>
`;
